contracts/multisig-prover/src/contract/execute.rs

```diff
@@ -149,6 +149,11 @@ fn make_verifier_set(
         return Err(ContractError::NotEnoughVerifiers.into());
     }
 
+    // Limit the number of signers for Stacks
+    if participants_with_pubkeys.len() > CLARITY_MAX_LEN_SIGNATURES as usize {
+        participants_with_pubkeys.truncate(CLARITY_MAX_LEN_SIGNATURES as usize);
+    }
```

Comment:
we should do some sort of sorting here before truncating, it might not be totally necessary but I am worried about the case where the service registry returns a vector of verifiers with non-deterministic ordering.

Reply:
It seems the service registry orders keys Ascending:
https://github.com/axelarnetwork/axelar-amplifier/blob/main/contracts/service-registry/src/contract/query.rs#L22

Not sure if we should add sorting here as well, to me it seems more like the job of the Service Registry itself.
This check would not even be necessary if the Service Registry takes into account the `max_num_verifiers` config, but that is outside of the scope of this discussion.

Let me know if it is fine like this or you still want to add ascending sorting by address here.

Reply:
hmm yes you're right, the order should be deterministic based on the service registry code. it wouldn't hurt to sort, but should be fine as is.